File: src/CorporateMessengerWithAgile/Application/Dto/Summary/KanbanBoardColumnSummaryDto.cs
using Domain.Entity;

namespace Application.Dto.Summary
{
    public class KanbanBoardColumnSummaryDto : BaseDto
    {
        public Guid TeamId { get; set; }
        public TaskItemStatus TaskStatus { get; set; }
        public int PositionOnBoard { get; set; }
        public string Title { get; set; } = null!;
    }
}

File: src/CorporateMessengerWithAgile/ReactApp/reactapp.client/src/models/entity/KanbanBoardColumnSummaryDto.ts
// src/models/entity/KanbanBoardColumnSummaryDto.ts
import { z } from 'zod';
import { BaseDtoSchema } from './BaseDto';
import { GuidSchema } from '../Guid';

export const KanbanBoardColumnSummaryDtoSchema = BaseDtoSchema.extend({
    teamId: GuidSchema,
    taskStatus: z.string().min(1),
    positionOnBoard: z.number(),
    title: z.string().min(1),
});

export type KanbanBoardColumnSummaryDto = z.infer<typeof KanbanBoardColumnSummaryDtoSchema>;
